sdks/python/build.gradle

```diff
@@ -25,75 +25,123 @@ apply plugin: "base"
 task test {}
 check.dependsOn test
 
-task setupTest {
+def envdir = "${project.buildDir}/gradleenv"
+
+task setupVirtualenv {
   doLast {
+    exec {
+      commandLine 'virtualenv', "${envdir}"
+    }
     exec {
       executable 'sh'
-      args '-c', 'which tox || pip install --user --upgrade tox'
+      args '-c', "source ${envdir}/bin/activate && pip install --upgrade tox"
     }
   }
+  outputs.files("$buildDir/.gradleenv/bin/tox")
 }
 
-task sdist {
+task sdist (dependsOn: 'setupVirtualenv') {
```

Comment:
nit: remove the space between task name and arguments since its a method call like `sdist(dependsOn: 'setupVirtualenv') {` here and below.

Reply:
done